protonmail/crypto.go

```diff
@@ -187,7 +213,34 @@ func (s signatureWriter) Close() error {
 	if err := sig.Serialize(s.encryptedData); err != nil {
 		return err
 	}
-	return s.encryptedData.Close()
+	if err := s.encryptedData.Close(); err != nil {
+		return err
+	}
+
+	var signatureErr error
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	spr, spw := io.Pipe()
```

Comment:
I don';t think we need a pipe nor a goroutine here. We should be able to directly serialize the signature to a `bytes.Buffer`.

Reply:
Sure, it was waaay to overcomplicated. Fixed it now.